hedyorg/hedy
# 📜 Add tags to mailchimp users

This PR adds the following functionality:
- when a user, who is subscribed to the newsletter, changes their country, the change is reflected in the Mailchimp subscriber profile.
- when a user creates or customizes a class, a corresponding tag is added to their Mailchimp subscriber profile.

Fixes partially #3995

**How to test**
Ensure you have access to the Mailchimp account of Hedy.
- Create a new teacher account and subscribe to the newsletter. Check that the Mailchimp subscriber has the tags 'teacher' and the country you selected.
- Create a class and do not customize it. Check that the Mailchimp subscriber has a 'created_class' tag.
- Customize the class and check that the Mailchimp subscriber has a new tag: 'customized_class'
- Go to the profile page of the teacher and change their country. Check that the Mailchimp subscriber has the country tag updated accordingly. The old country, if any, should be removed and the new country, if any, should be added.
- Go to the profile page of the teacher and change their email. Check that there is a Mailchimp subscriber with the new email address and that all tags from the old subscriber are present. Check that the old Mailchimp subscriber is archived.

Note that currently you cannot subscribe to the newsletter after you signed up. So you cannot check if a teacher with existing customized classes gets the right tags when they subscribed.

To consider:
All functions for managing subscriptions are blocking but we do not need their output to continue with the response and the calls to Mailchimp are not retried. Is it worth investing in these for this feature?
